conans/client/cmd/new.py

```diff
@@ -236,13 +237,41 @@ def test(self):
 
 """
 
+def _render_template(text, name, version, package_name):
+    context = {'name': name, 
+               'version': version, 
+               'package_name': package_name,
+               'conan_version': client_version}
+    t = Template(text, keep_trailing_newline=True)
+    return t.render(**context)
+
+def _get_files_from_template_dir(template_dir, name, version, package_name):
+    if (not os.path.isfile(os.path.join(template_dir, "conanfile.py")) 
+        and not os.path.isfile(os.path.join(template_dir, "conanfile.txt"))):
+        raise ConanException("Template is missing a recipe file: {}".format(template_dir))
+
+    files = []
+    for d, _, fs in os.walk(template_dir):
+        for f in fs:
+            rel_d = os.path.relpath(d, template_dir)
+            rel_f = os.path.join(rel_d, f)
+            files.append(rel_f)
+
+    out_files = dict()
+    for f in files:
+        f_path = os.path.join(template_dir, f)
+        rendered_path = _render_template(f, name=name, version=version, package_name=package_name)
+        rendered_file = _render_template(load(f_path), name=name, version=version, package_name=package_name)
+        out_files[rendered_path] = rendered_file
+
+    return out_files
 
 def cmd_new(ref, header=False, pure_c=False, test=False, exports_sources=False, bare=False,
             visual_versions=None, linux_gcc_versions=None, linux_clang_versions=None,
             osx_clang_versions=None, shared=None, upload_url=None, gitignore=None,
             gitlab_gcc_versions=None, gitlab_clang_versions=None,
             circleci_gcc_versions=None, circleci_clang_versions=None, circleci_osx_versions=None,
-            template=None, cache=None):
+            template=None, template_dir=None, cache=None):
```

Comment:
Seems ``template_dir`` argument not used here?

Reply:
Fixed